R/mail.R

```diff
@@ -6,110 +6,249 @@
+    }
+
+    # Add the downloaded file to the list
+    downloaded_files <- c(downloaded_files, dest)
+  }
 
+  # Return downloaded files
+  return(downloaded_files)
 }
 
 
-#' Convert pipermail archive files (.txt and .txt.gz) into an mbox format for use with \code{\link{parse_mbox}}
-#' @param filelist A vector of pipermail archive files from \code{\link{download_pipermail}}
-#' @return Returns `output`, the name of the resulting .mbox file in the current working directory
+#' Refresh mbox files downloaded via pipermail
+#' Uses the adopted file name convention by \code{\link{download_pipermail}} to identify
+#' the latest downloaded mbox. It deletes this file, then redownloads it along with all future months
+#' up to the current real-life month.
+#' If the directory is empty, then it downloads all pipermail files (as mbox files) via \code{\link{download_pipermail}}
+#' @param mailing_list The name of the mailing list being downloaded
+#' @param start_year_month The year and month of the first file to be downloaded
+#' @param save_folder_path The folder path in which all the downloaded pipermail files will be stored
+#' @return Returns `downloaded_files`, a vector of the downloaded files in the current working directory
 #' @export
-convert_pipermail_to_mbox <- function(filelist) {
+refresh_pipermail <- function(mailing_list, start_year_month, save_folder_path) {
 
-  #at to @ replace function
-  pipermail_atreplacer <- function(string) {
+  # Create directory if it does not exist
+  if (!dir.exists(save_folder_path)) {
+    dir.create(save_folder_path, recursive = TRUE)
+  }
 
-    rstring <- sub(" at ", "@", string)
+  # Check if the folder is empty
+  files_in_folder <- list.files(save_folder_path, pattern = "kaiaulu_\\d{6}\\.mbox$")
+  if (length(files_in_folder) == 0) {
+    # If empty, download from start_year_month to the current month
+    end_year_month <- format(Sys.Date(), "%Y%m")
+    cat("Folder is empty. Downloading from", start_year_month, "to", end_year_month, "\n")
+    download_pipermail(mailing_list, start_year_month, end_year_month, save_folder_path)
+    return(NULL)
+  }
+  # If folder is not empty, find the most recent month
+  year_months <- gsub("kaiaulu_(\\d{6})\\.mbox$", "\\1", files_in_folder)
+  recent_month <- max(year_months)
+
+  # Delete the most recent file
+  recent_file <- file.path(save_folder_path, paste0("kaiaulu_", recent_month, ".mbox"))
+  if (file.exists(recent_file)) {
+    file.remove(recent_file)
+    cat("Deleted the most recent file:", recent_file, "\n")
+  }
 
-    return(rstring)
+  # Redownload from the most recent month to the current real-life month
+  end_year_month <- format(Sys.Date(), "%Y%m")
+  cat("Redownloading from", recent_month, "to", end_year_month, "\n")
+  download_pipermail(mailing_list, recent_month, end_year_month, save_folder_path)
+}
 
-  }
 
-  output <- "output.mbox"
+#' Process .gz files in a folder, unzip and convert them to .mbox
```

Comment:
Missing title. May want to check other function docs in this PR to make sure they have the proper format.